### DISCUSSION TITLE
SBML to mrgsolve

### DISCUSSION THREAD

**Discussion by mhismail** [metrumresearchgroup/mrgsolve#820] [Q&A]
Hi @kylebaron , I was wondering if there are any open-source SBML to mrgsolve translators available, or generally any tips for programmatically going from SBML to mrgsolve. Appreciate any insights.

Thanks.

**Comment by kylebaron** [comment]
Hi @mhismail - 

Nothing open source, but I've done it several times.  Most of the code is the same but IMO little tweaks are needed pretty regularly since the structure / assumptions can vary depending on where the SBML is coming from. 

I have the code to handle this SBML:
https://www.nature.com/articles/s41540-017-0016-1

I can put that somewhere as an example if that'd be helpful. 

Kyle

**Reply by mhismail** [reply]
Hi @kylebaron, that would be really helpful if you could share that as an example. Thank you!

**Reply by KGmax** [reply]
Can you share it, or describe how to make this conversion? It would help me tremendously. SMBL is too complicated and converting it to mrgsolve would takes a lot of time for a beginner like me.

**Reply by KGmax** [reply]
In the article (doi:10.1002/psp4.12467) you write, "The translation code was written in R and utilized an R interface to libSBML as linked from https://sbml.org.". But how was this done? I've been trying all day but haven't made any progress.

**Comment by kylebaron** [comment]
@mhismail You might look at this too:

https://github.com/sn248/r2sbml

Written by @sn248; it is not on cran but might be less monkey business than what I've written to interface with libSBML (it probably **is** less monkey business).

**Reply by mhismail** [reply]
Thanks @kylebaron, I was able to get things running with the other script you shared. I'll definitely give this package a try next time.

**Comment by Mutaz94** [comment]
Hi @kylebaron, this is interesting since I was looking around. Did you share libSBML code? I was looking around but without luck!

**Reply by kylebaron** [reply]
Hi @Mutaz94 - I never got something to the point where I wanted to share publicly. Kyle
